addons/remote-write-ingress.libsonnet

```diff
@@ -0,0 +1,93 @@
+function(config) {
+  assert std.objectHas(config, 'prometheus') : 'prometheus required',
+  assert std.objectHas(config.prometheus, 'DNS') : 'prometheus.DNS required',
+  assert std.objectHas(config.prometheus, 'nodePort') : 'prometheus.nodePort required',
+  assert std.isNumber(config.prometheus.nodePort) : 'prometheus.nodePort should be a number',
+  assert std.objectHas(config.prometheus, 'GCPExternalIpAddress') : 'prometheus.GCPExternalIpAddress required',
+  assert std.objectHas(config.prometheus, 'BasicAuthSecret') : 'prometheus.BasicAuthPassword required',
+
+
+  prometheus+: {
+    certificate: {
+      apiVersion: 'cert-manager.io/v1',
+      kind: 'Certificate',
+      metadata: {
+        name: 'prometheus-remote-write',
+        namespace: config.namespace,
+      },
+      spec: {
+        dnsNames: [
+          config.prometheus.DNS,
+        ],
+        issuerRef: {
+          kind: 'ClusterIssuer',
+          name: 'letsencrypt-issuer',
+        },
+        secretName: 'prometheus-remote-write-certificate',
+      },
+    },
+
+    secret: {
+        apiVersion: 'v1',
+        kind: 'Secret',
+        metadata: {
+            name: 'prometheus-remote-write-basic-auth',
+            namespace: config.namespace,
+            labels: $.prometheus.service.metadata.labels,
+        },
+        spec: {
+            auth: std.base64(config.prometheus.BasicAuthPassword),
```

Comment:
Doesn't basic auth require user and password? 🤔

Reply:
see the update in the generate. The string contains both in a "user:hashed_password" format